bases/renku_data_services/background_jobs/utils.py

```diff
@@ -0,0 +1,60 @@
+"""Different utility functions for background jobs."""
+
+import logging
+
+from authzed.api.v1.permission_service_pb2 import ReadRelationshipsRequest, RelationshipFilter
+
+from renku_data_services.authz.authz import Authz, ResourceType, _Relation
+from renku_data_services.background_jobs.config import SyncConfig
+from renku_data_services.message_queue.avro_models.io.renku.events import v2
+from renku_data_services.message_queue.converters import EventConverter
+
+
+async def sync_user_namespaces(config: SyncConfig) -> None:
+    """Lists all user namespaces in the database and adds them to Authzed and the event queue."""
+    authz = Authz(config.authz_config)
+    user_namespaces = await config.group_repo._get_user_namespaces()
+    logging.info(f"Start syncing {len(user_namespaces)} to the authorization DB and message queue")
+    for user_namespace in user_namespaces:
+        events = EventConverter.to_events(user_namespace, v2.UserAdded)
+        authz_change = authz._add_user_namespace(user_namespace.namespace)
+        session = config.session_maker()
+        tx = session.begin()
+        await tx.start()
+        try:
+            await authz.client.WriteRelationships(authz_change.apply)
+            for event in events:
+                await config.event_repo.store_event(session, event)
+        except Exception as err:
+            logging.error(f"Failed to sync user namespace {user_namespace} because {err}")
+            await tx.rollback()
```

Comment:
do we also need to try to authz_change.undo here?

Reply:
I do not want to do that because if the authz update succeeded I will leave it in. It is acceptable to have something in the authz DB but not in the message queue but not vice versa.